needy/cache.py

```diff
@@ -0,0 +1,29 @@
+
+class Cache:
+    def __init__(self, configuration={}):
+        self.__configuration = configuration
+
+    @staticmethod
+    def type():
+        '''type of cache'''
+        raise NotImplementedError('type')
+
+    def description(self):
+        '''uri, path, or other configuration description'''
+        raise NotImplementedError('description')
+
+    def configuration(self):
+        '''configuration dictionary to serialize to configuration files'''
+        return self.__configuration
+
+    def archive(self, source, key):
+        '''make files and folders at source retrievable with the given key'''
+        raise NotImplementedError('archive')
+
+    def unarchive(self, destination, key):
```

Comment:
I'm suggesting two abstract methods to be supplied by implementations of the Cache class, and two methods of the Cache class itself that would utilize those methods. The goal here is for subclasses of Cache to only have to provide the essential building blocks we need.

```
class Cache:
    def load_directory(self, key, destination):
        self.load_file(key, some_temporary_location)
        unpack_tarball(some_temporary_location, destination)

    def store_directory(self, key, source):
        pack_tarball(source, some_temporary_location)
        self.store_file(key, some_temporary_location)

    def load_file(key, destination):
        raise NotImplementedError

    def store_file(key, source):
        raise NotImplementedError
```

Cache implementations then implement those two methods, don't duplicate tarballing code, and provide more granular functionality that we can reuse for more potential needs.